sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3831,6 +3831,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val PIPE_OPERATOR_ALLOW_AGGREGATE_IN_SELECT =
+    buildConf("spark.sql.pipeOperator.allowAggregateInSelect")
+    .doc("When true, aggregate functions can be used in |> SELECT and other pipe operator " +
+      "clauses without requiring the |> AGGREGATE keyword. When false, aggregate functions " +
+      "must be used exclusively with the |> AGGREGATE clause for proper aggregation semantics.")
+    .version("4.1.0")
+    .booleanConf
+    .createWithDefault(true)
```

Comment:
Please turn it off by default in Apache Spark 4.1.0. You can enable it later for Apache Spark 4.2.0.